code/game/objects/items/stacks/sheets/sheet_types.dm

```diff
@@ -253,19 +253,15 @@ GLOBAL_LIST_INIT(cardboard_recipes, list ( \
 	amount = 50
 
 /obj/item/stack/sheet/cardboard/attackby(obj/item/I, mob/user, params)
-	if(!istype(I, /obj/item/stamp/clown))
-		return ..()
-	if(istype(loc, /obj/item/storage))
-		to_chat(user,"<span class='notice'>Take the [src] out first!</span>")
-		return ..()
-	var/turf/T = get_turf(src.loc)
-	playsound(src, 'sound/items/bikehorn.ogg', 50, 1, -1)
-	to_chat(user, "<span class='notice'>You stamp the cardboard! Its a clown box! Honk!</span>")
-	new/obj/item/storage/box/clown(drop_location(T))
-	if ((src.amount > 1) && (user.is_holding(src))) //prevent a bug when holding stack
-		amount-- //you also keep the stack in your hand
-	else
-		use(1)
+	if(istype(I, /obj/item/stamp/clown) && !istype(loc, /obj/item/storage))
+		var/atom/droploc = drop_location()
+		if(use(1))
+			playsound(I, 'sound/items/bikehorn.ogg', 50, 1, -1)
+			to_chat(user, "<span class='notice'>You stamp the cardboard! Its a clown box! Honk!</span>")
+			if (amount == 0) new/obj/item/storage/box/clown(droploc) //bugfix
+			else new/obj/item/storage/box/clown(drop_location())
```

Comment:
I don't understand the purpose of this if and else statement

What happens if instead of this whole if statement, you simply have `new/obj/item/storage/box/clown(droploc) `?

Reply:
tl;dr? drop_location() doesn't work with single stack.

If we use box/clown(droploc) for both, they'd both act the same.
We could save a line by removing 262, I guess.